libcaf_net/caf/net/http/request_header.cpp

```diff
@@ -13,6 +13,46 @@ namespace {
 
 constexpr std::string_view eol = "\r\n";
 
+/// Validate the request target part according to RFC9112.
+caf::expected<uri> validate_request_target(http::method method,
```

Comment:
For a function named `validate`, I would expect a `bool` or maybe `error` return type. This actually tries to parse its input, so maybe `parse_request_target` instead?